pkg/instrumentation/podmutator.go

```diff
@@ -427,6 +430,9 @@ func (pm *instPodMutator) Mutate(ctx context.Context, ns corev1.Namespace, pod c
 
 func (pm *instPodMutator) getInstrumentationInstance(ctx context.Context, ns corev1.Namespace, pod corev1.Pod, instAnnotation string) (*v1alpha1.Instrumentation, error) {
 	instValue := annotationValue(ns.ObjectMeta, pod.ObjectMeta, instAnnotation)
+	if len(instValue) == 0 && pm.isDynamicInstrumentationEnabled(strings.TrimPrefix(instAnnotation, annotationInjectPrefix), pod) {
```

Comment:
Can this annotation value return "false"? If so maybe we can reuse the check below?

Reply:
it it's "false", we should respect that and not apply dynamic configuration